app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsTypeItem.kt

```diff
@@ -0,0 +1,56 @@
+package org.wikipedia.suggestededits
+
+import android.content.Context
+import android.content.res.ColorStateList
+import android.os.Build
+import android.util.AttributeSet
+import android.view.View
+import android.widget.LinearLayout
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.core.content.ContextCompat
+import androidx.core.widget.ImageViewCompat
+import kotlinx.android.synthetic.main.item_suggested_edits_type.view.*
+import org.wikipedia.R
+import org.wikipedia.suggestededits.Contribution.Companion.ALL_EDIT_TYPES
+import org.wikipedia.util.ResourceUtil
+
+class SuggestedEditsTypeItem constructor(context: Context, attrs: AttributeSet? = null) : LinearLayout(context, attrs) {
+
+    var editType: Int = ALL_EDIT_TYPES
+    private var callback: Callback? = null
+
+    interface Callback {
+        fun onClick(view: SuggestedEditsTypeItem)
+    }
+
+    init {
+        View.inflate(context, R.layout.item_suggested_edits_type, this)
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            foreground = AppCompatResources.getDrawable(context, ResourceUtil.getThemedAttributeId(context, R.attr.selectableItemBackground))
+        }
+        setOnClickListener {
+            if (callback != null) {
+                callback!!.onClick(this)
```

Comment:
Kotlin allows us to do `callback?.onClick()` without needing to do an explicit null check ourselves.